addon/utils/reopen-route.js

```diff
@@ -1,24 +1,23 @@
 import Ember from 'ember';
 import arraySwap from 'ember-redirect/utils/array-swap';
+import { lookup, register } from 'ember-redirect/utils/container';
 
 export default function(routeName, options, instance) {
-  var routeContainerKey = `route:${routeName}`;
-  var routeObject       = instance.container.lookup(routeContainerKey);
+  let routeContainerKey = `route:${routeName}`;
+  let routeObject       = lookup(instance, routeContainerKey);
 
   if (!routeObject) {
     routeObject = Ember.Route.extend({});
-    instance.registry.register(routeContainerKey, routeObject, { singleton: false });
+    register(instance, routeContainerKey, routeObject, { singleton: false });
   }
 
-  Ember.assert(`Could not find a route named: ${routeName}`, routeObject);
```

Comment:
I believe it had to do with if you had a route defined in router.js but did not have an actual route.js file to back it.